R/backend-mssql.R

```diff
@@ -177,6 +177,13 @@
 
   )}
 
+#' @export
+`sql_escape_raw.Microsoft SQL Server` <- function(con, x) {
+  # SQL Server binary constants should be prefixed with 0x
+  # https://docs.microsoft.com/en-us/sql/t-sql/data-types/constants-transact-sql?view=sql-server-ver15#binary-constants
+  paste0(c("0x", format(x)), collapse = "")
```

Comment:
I benchmarked performance of some variants and a C/C++ implementation in https://github.com/r-lib/rlang/issues/978. No need to act here.